src/game/Pet.cpp

```diff
@@ -143,15 +143,6 @@ bool Pet::LoadPetFromDB(Player* owner, uint32 petentry, uint32 petnumber, bool c
     uint32 summon_spell_id = fields[21].GetUInt32();
     SpellEntry const* spellInfo = sSpellStore.LookupEntry(summon_spell_id);
 
-    bool is_temporary_summoned = spellInfo && GetSpellDuration(spellInfo) > 0;
```

Comment:
My statement is that we never need to save temporary summons to the db, and never need to fetch them from there again. this is because... yes.. there actually are no timed summons that are controlled in vanilla, no. Can't think of a single one.